Reconstruction/RecCalorimeter/src/components/CreateCaloClustersSlidingWindow.cpp

```diff
@@ -213,18 +225,55 @@ StatusCode CreateCaloClustersSlidingWindow::execute() {
   double radius = m_towerTool->radiusForPosition();
   auto edmClusters = m_clusters.createAndPut();
   for (const auto clu : m_preClusters) {
-    auto edmCluster = edmClusters->create();
-    auto& edmClusterCore = edmCluster.core();
-    edmClusterCore.position.x = radius * cos(clu.phi);
-    edmClusterCore.position.y = radius * sin(clu.phi);
-    edmClusterCore.position.z = radius * sinh(clu.eta);
-    edmClusterCore.energy = clu.transEnergy * cosh(clu.eta);
-    debug() << "Cluster eta: " << clu.eta << " phi: " << clu.phi << "x: " << edmClusterCore.position.x << " y "
-            << edmClusterCore.position.y << " z " << edmClusterCore.position.z << " energy " << edmClusterCore.energy
-            << endmsg;
-    if (m_saveCells) {
-      // fins cells that belong to the cluster
-      m_towerTool->matchCells(clu.eta, clu.phi, halfEtaWin, halfPhiWin, edmCluster);
+    float clusterEnergy = clu.transEnergy * cosh(clu.eta);
+    // apply energy sharing correction (if flag set to true)
+    if (m_energySharingCorrection) {
+      int idEtaCl = m_towerTool->idEta(clu.eta);
+      int idPhiCl = m_towerTool->idPhi(clu.phi);
+      std::vector<std::vector<float>> sumEnergySharing;
+      sumEnergySharing.assign(2 * halfEtaFin + 1, std::vector<float>(2 * halfPhiFin + 1, 0));
+      // loop over all clusters and check if they have any tower in common with our current cluster
+      for (const auto iclu : m_preClusters) {
+        int idEtaIcl = m_towerTool->idEta(iclu.eta);
+        int idPhiIcl = m_towerTool->idPhi(iclu.phi);
+        if (idEtaCl != idEtaIcl && idPhiCl != idPhiIcl) {
+          if (abs(idEtaIcl - idEtaCl) <= halfEtaFin && abs(idPhiIcl - idPhiCl) <= halfPhiFin) {
+            int iEtaShareMin = (idEtaIcl < idEtaCl) ? idEtaCl - halfEtaFin : idEtaIcl - halfEtaFin;
+            int iEtaShareMax = (idEtaIcl < idEtaCl) ? idEtaIcl + halfEtaFin : idEtaCl + halfEtaFin;
+            int iPhiShareMin = (idPhiIcl < idPhiCl) ? idPhiCl - halfPhiFin : idPhiIcl - halfPhiFin;
+            int iPhiShareMax = (idPhiIcl < idPhiCl) ? idPhiIcl + halfPhiFin : idPhiCl + halfPhiFin;
```

Comment:
I think this would be more readable with `std::min` / `std::max`. Also I don't get the meaning of the suffix `Icl` :)